src/croniter/croniter.py

```diff
@@ -816,11 +849,12 @@ def _expand(cls, expr_format, hash_id=None, second_at_beginning=False, from_time
                         pass
 
                     if t in cls.LOWMAP[i] and not (
-                        # do not support 0 as a month either for classical 5 fields cron
-                        # or 6fields second repeat form
+                        # do not support 0 as a month either for classical 5 fields cron,
+                        # 6fields second repeat form or 7 fields year form
                         # but still let conversion happen if day field is shifted
-                        (i in [2, 3] and len(expressions) == 5) or
-                        (i in [3, 4] and len(expressions) == 6)
+                        (i in [DAY_FIELD, MONTH_FIELD] and len(expressions) == UNIX_CRON_LEN) or
+                        (i in [MONTH_FIELD, DOW_FIELD] and len(expressions) == SECOND_CRON_LEN) or
+                        (i in [DAY_FIELD, MONTH_FIELD, DOW_FIELD] and len(expressions) == YEAR_CRON_LEN)
```

Comment:
Well those tests are maybe broken (not from your, initially), i have to rethink about this...